examples/Custom_Hooks/campaign_stopping.py

```diff
@@ -0,0 +1,238 @@
+## Campaign Stopping
+
+# This example demonstrates how to use custom hook mechanics to stop unpromising
+# campaigns based on a simple Probability of Improvement (PI) criterion.
+# This is based on the "full lookup" example [here](../Backtesting/full_lookup.md).
+
+### Imports
+
+import os
+import warnings
+
+import pandas as pd
+import seaborn as sns
+import torch
+
+from baybe import Campaign
+from baybe.acquisition import ProbabilityOfImprovement
+from baybe.exceptions import UnusedObjectWarning
+from baybe.objectives import SingleTargetObjective
+from baybe.objectives.base import Objective
+from baybe.parameters import (
+    NumericalDiscreteParameter,
+    SubstanceParameter,
+)
+from baybe.recommenders import (
+    BotorchRecommender,
+    RandomRecommender,
+    TwoPhaseMetaRecommender,
+)
+from baybe.searchspace import SearchSpace, SearchSpaceType
+from baybe.simulation import simulate_experiment, simulate_scenarios
+from baybe.targets import NumericalTarget
+from baybe.utils import register_hooks
+from baybe.utils.dataframe import to_tensor
+from baybe.utils.plotting import create_example_plots
+from baybe.utils.random import set_random_seed
+
+### Temporary
+warnings.filterwarnings(
+    "ignore", category=UnusedObjectWarning, message="explicit objective"
+)
+warnings.filterwarnings("ignore", category=DeprecationWarning)
+
+### Settings
+
+# For the full simulation, we need to define some additional settings.
+# These are the number of Monte Carlo runs, the number of experiments to be conducted
+# per run and number of campaigns we run with the stopping criterion.
+# We also fix the random seed for reproducibility.
+
+SMOKE_TEST = "SMOKE_TEST" in os.environ
+
+N_DOE_ITERATIONS = 2 if SMOKE_TEST else 20
+N_MC_ITERATIONS = 2 if SMOKE_TEST else 50
+N_STOPPED_CAMPAIGNS = 2 if SMOKE_TEST else 5
+BATCH_SIZE = 1
+set_random_seed(1337)
+
+### Lookup Functionality and Parameter Data
+
+# This loads the dataframe used for looking up measurements in the simulation.
+
+try:
+    lookup = pd.read_excel("./../Backtesting/lookup.xlsx")
+except FileNotFoundError:
+    try:
+        lookup = pd.read_excel("examples/Backtesting/lookup.xlsx")
+    except FileNotFoundError as e:
+        print(e)
+
+# Then we define the substance categories and SMILES representing their molecular structure.
+# Note that we need to ensure that the names fit the names in the provided .xlsx file!
```

Comment:
Can we delete this line and the general explanations on this example regarding the search space and just refer to the `full_lookup` example?

Reply:
added one more reference to that and deleted the two sentences